### PR TITLE
VueUI Titlebars, Resizing, and Scrolling other edition

### PR DESCRIPTION
**Pull request by Karolis2011**
Thanks to @JohnWildkins for most of work, and his original PR #10528

Modular computers look like: 
imageimage
image
image

### CODE REVIEW

File: code/modules/vueui/ui.dm [hunk 1]
```diff
@@ -178,6 +174,13 @@ main ui datum.
 	sdata["status"] = status
 	sdata["title"] = title
 	sdata["wtime"] = world.time
+	sdata["debug"] = user && check_rights(R_DEV, FALSE, user=user)
```
Comment (JohnWildkins) [inline code/modules/vueui/ui.dm:177]:
```suggestion
```

This is set twice here

**Review by JohnWildkins — CHANGES_REQUESTED** [review]
There's nothing super syntactically wrong and the JS code for dragging is solid, which is great. The styling is pretty off, though. Compare your nano-dark header to mine:

yours
mine

I feel that the latter is cleaner; there's no awkward gap between the headers and the spacebar, the computer's header is distinct from the main titlebar, and the elements on the top bar are all centered with respect to each other. (As I said on discord, I'm also not a fan of how thick the X button is, and doubly so that it isn't a hover-over with a changing background).

This PR is also missing the debug button functionality; right now it seems to load the debug menu for everyone(?) which I'm assuming is something WIP. The hacky focusing events I was using to maintain focus on the window are also gone, which is resaonable, but it means that you now lose focus on the window if you resize it. 

It's obviously not my call but I'd prefer if the backend/JS changes here were simply brought onto my PR, rather than the other way around.

**Comment by Karolis2011** [comment]
!review

**Comment by Karolis2011** [comment]
Focus issues have been addressed.
Ui closing have been handled in a way that allows any UI to be closed.
Now if you have UIDEBUG enabled behavior is same as old, but if you have rights you get debug button that allows to enable debug view.
Also odd gaps have been addressed.
image
image

**Comment by skull132** [comment]
PR rejected in favour of Wildkins'.

I prefer the look of the other one. Plus, you were non-communicative over the course of handling these matters, ergo, we prefer the side which actually talked.

I think the wonkiness of dragging is the only issue Wildkins' PR doesn't yet fix, so you're free to port an improved JS backend to master.

**Pull request closed without merging**